ambry-router/src/main/java/com/github/ambry/router/PutOperation.java

```diff
@@ -2116,7 +2124,8 @@ private void finalizeMetadataChunk() {
             .stream()
             .map(p -> new Pair<>(p.getFirst().toString(), p.getSecond()))
             .collect(Collectors.toList()));
-        PutBlobMetaInfo putBlobMetaInfoObj = new PutBlobMetaInfo(orderedChunkList, reservedMetadataChunkId.getID());
+        // S3 Part Upload, we don't reserve metadata chunk id for each data chunk. Refer to resolveReservedMetadataId.
+        PutBlobMetaInfo putBlobMetaInfoObj = new PutBlobMetaInfo(orderedChunkList, null);
```

Comment:
Would we have to ignore the reserved metadata Id at the time of stitching too? https://github.com/linkedin/ambry/blob/0b197de3b1b2db94e71afd261534c327702f8b26/ambry-router/src/main/java/com/github/ambry/router/PutOperation.java#L1944

Reply:
Added handling for the stitch command.